src/llm/apis/openai_completions.cpp

```diff
@@ -495,52 +495,27 @@ absl::Status OpenAIChatCompletionsHandler::parseCommonPart(std::optional<uint32_
         request.numReturnSequences = it->value.GetUint();
     }
 
-    // Speculative decoding specific parameters
+    // Assisted decoding specific parameters
 
     auto numAssistantTokensIt = doc.FindMember("num_assistant_tokens");
     auto assistantConfidenceThresholdIt = doc.FindMember("assistant_confidence_threshold");
+    auto maxNgramSizeIt = doc.FindMember("max_ngram_size");
 
     bool numAssistantTokensItHasValue = (numAssistantTokensIt != doc.MemberEnd() && !numAssistantTokensIt->value.IsNull());
     bool assistantConfidenceThresholdItHasValue = (assistantConfidenceThresholdIt != doc.MemberEnd() && !assistantConfidenceThresholdIt->value.IsNull());
+    bool maxNgramSizeItHasValue = (maxNgramSizeIt != doc.MemberEnd() && !maxNgramSizeIt->value.IsNull());
 
-    if (isSpeculativePipeline) {
-        if (!numAssistantTokensItHasValue && !assistantConfidenceThresholdItHasValue)
```

Comment:
we get rid of `numAssistantTokensItHasValue` completely? how does compiler let it through, if `numAssistantTokensItHasValue` is now unused?

Reply:
same with other vars

Reply:
It is used. Github does not display it in a very friendly way (they still there in between red, removed parts)